state/storage.go

```diff
@@ -206,6 +209,8 @@ func (s *Storage) MetadataForAllRooms(txn *sqlx.Tx, tempTableName string, result
 		metadata.LatestEventsByType[parsed.Get("type").Str] = eventMetadata
 		// it's possible the latest event is a brand new room not caught by the first SELECT for joined
 		// rooms e.g when you're invited to a room so we need to make sure to set the metadata again here
+		// TODO: is this accurate now that we explicitly call NewRoomMetadata above
+		//       when handling invites?
```

Comment:
Is what accurate? The room ID? It should be if we're setting it in NewRoomMetadata?

Reply:
Sorry "this" means "the comment above". I.e. _is_ it possible for us to arrive at line 202 now?

My thinking is "maybe not, but I don't have the time to investigate further and it's not broken as-is".